src/benchmarklib/listenable.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <json.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace opossum {
+
+enum class Event {
+  ItemRunStarted,
+  ItemRunFinished,
+  CreateReport,
```

Comment:
Having one global list of events seems to be a smell. How about letting each Listenable create its own `enum class` and use CRTP (`: public Listenable<BenchmarkRunner`)?

Reply:
I will look into that